internal/uvm/scsi.go

```diff
@@ -323,9 +341,40 @@ func (uvm *UtilityVM) addSCSIActual(ctx context.Context, hostPath, uvmPath, atta
 	}
 
 	if err := uvm.modify(ctx, SCSIModification); err != nil {
-		return nil, fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
+		return fmt.Errorf("uvm::AddSCSI: failed to modify utility VM configuration: %s", err)
 	}
-	return sm, nil
+	return nil
+}
+
+// allocateSCSIDevice grants vm access to hostpath and increments the ref count of an existing scsi
+// device or allocates a new one if not already present.
+func (uvm *UtilityVM) allocateSCSIDevice(ctx context.Context, hostPath, uvmPath string, vmAccess VMAccessType) (*SCSIMount, bool, error) {
```

Comment:
Nit: Small comment on what the bool being returned is for so we don't have to go to the callers of this function to see 'existed'.

Reply:
Done.